### ISSUE TITLE
ember-simple-auth example is out of date

### ISSUE THREAD

**Issue by myasonik** [ember-graphql/ember-apollo-client#381]
The [`ember-simple-auth`](https://github.com/simplabs/ember-simple-auth) example for adding headers uses an API that's no longer supported in 3.0: `this.session.authorize`. 

Per their [upgrade guide](https://github.com/simplabs/ember-simple-auth/blob/master/guides/upgrade-to-v3.md#refactor-ember-data-adapters-to-remove-use-of-authorizers), users should now use `this.session.data.authenticated` to get whatever info they may need.

**Comment by myasonik** [comment]
I've validated that this (lightly simplified) works in a personal project:

```
import ApolloService from 'ember-apollo-client/services/apollo';
import { inject as service } from '@ember/service';
import { setContext } from '@apollo/client/link/context';

class OverriddenApollo extends ApolloService {
  @service
  session;

  link() {
    let httpLink = super.link();

    let authLink = setContext((request, context) => {
      return this._runAuthorize(request, context);
    });
    return authLink.concat(httpLink);
  }

  _runAuthorize() {
    if (!this.session.isAuthenticated) {
      return {};
    }
    return new Promise(success => {
      success({ headers: this.session.data.authenticated.headers });
    });
  }
}
```

**Comment by josemarluedke** [comment]
Fixed in #377.
